xonotic2.patch

```diff
@@ -0,0 +1,424 @@
++	# install-links
++	$(INSTALL_PROGRAM) xonotic-linux-sdl.sh $(DESTDIR)$(BASEDIR)/xonotic-linux-sdl.sh
++	$(INSTALL) -d $(DESTDIR)$(bindir)
++	ln -snf $(BASEDIR)/xonotic-linux-sdl.sh $(DESTDIR)$(bindir)/xonotic-sdl
++endif
++
++.PHONY: install-engine-server
++install-engine-server: server
++ifdef DP_FS_BASEDIR # path for distro package builds that define $BASEDIR / $DP_FS_BASEDIR
++	$(INSTALL_PROGRAM) source/darkplaces/darkplaces-dedicated $(DESTDIR)$(bindir)/xonotic-dedicated
++else # end users aren't expected to `make install` but if they do this path makes their install functional
++	$(INSTALL_PROGRAM) source/darkplaces/darkplaces-dedicated $(DESTDIR)$(BASEDIR)/xonotic-dedicated
++	# install-links
++	$(INSTALL_PROGRAM) xonotic-linux-dedicated.sh $(DESTDIR)$(BASEDIR)/xonotic-linux-dedicated.sh
++	$(INSTALL) -d $(DESTDIR)$(bindir)
++	ln -snf $(BASEDIR)/xonotic-linux-dedicated.sh $(DESTDIR)$(bindir)/xonotic-dedicated
++endif
+ 
++# Flathub requires these file names to be changed, which requires editing files that reference them.
++# No file extensions in the values because desktop files omit them from Icon=
++FILENAME_ICON_CLIENT ?= xonotic
++FILENAME_DESKTOP_CLIENT ?= xonotic
++.PHONY: install-desktop-client
++install-desktop-client:
++	$(INSTALL_DATA) misc/logos/xonotic_icon.svg $(DESTDIR)$(datarootdir)/icons/hicolor/scalable/apps/$(FILENAME_ICON_CLIENT).svg
++	$(INSTALL_DATA) misc/logos/xonotic.desktop $(DESTDIR)$(datarootdir)/applications/$(FILENAME_DESKTOP_CLIENT).desktop
++	$(INSTALL_DATA) misc/logos/org.xonotic.Xonotic.metainfo.xml $(DESTDIR)$(datarootdir)/metainfo/org.xonotic.Xonotic.metainfo.xml
++	sed -i 's/Icon=xonotic/Icon=$(FILENAME_ICON_CLIENT)/' $(DESTDIR)$(datarootdir)/applications/$(FILENAME_DESKTOP_CLIENT).desktop
++	sed -i 's/<launchable type=\"desktop-id\">xonotic.desktop<\/launchable>/<launchable type=\"desktop-id\">$(FILENAME_DESKTOP_CLIENT).desktop<\/launchable>/' $(DESTDIR)$(datarootdir)/metainfo/org.xonotic.Xonotic.metainfo.xml
++
++.PHONY: install-desktop-server
++install-desktop-server: # TODO https://gitlab.com/xonotic/xonotic/-/issues/216
++
++.PHONY: install-doc-client
++install-doc-client:
++	$(INSTALL) -d $(DESTDIR)$(docdir)
++	cp -R Docs/* $(DESTDIR)$(docdir)/
++
++.PHONY: install-doc-server
++install-doc-server:
++	$(INSTALL) -d $(DESTDIR)$(docdir)
++	cp -R server $(DESTDIR)$(docdir)/
+diff --git a/misc/logos/org.xonotic.Xonotic.metainfo.xml b/misc/logos/org.xonotic.Xonotic.metainfo.xml
+new file mode 100644
+index 0000000..35fcb54
+--- /dev/null
++++ b/misc/logos/org.xonotic.Xonotic.metainfo.xml
+@@ -0,0 +1,106 @@
++<?xml version="1.0" encoding="UTF-8"?>
++<component type="desktop-application">
++  <id>org.xonotic.Xonotic</id>
++  <project_license>GPL-3.0-or-later</project_license>
++  <metadata_license>CC0-1.0</metadata_license>
++  <provides>
++    <binary>xonotic-sdl</binary>
++  </provides>
++  <launchable type="desktop-id">xonotic.desktop</launchable>
```

Comment:
Do not forget to fix the desktop file name here. :-)

It can be fixed using sed in the xonotic module _build-commands_ for consistency with the upstream file instead of directly modifying the (temporary) patch.

Reply:
Oh, it is already in the Makefile. :-) Cool! Feel free to ignore my previous comment.

Reply:
It's done in the Makefile updates https://gitlab.com/xonotic/xonotic/-/merge_requests/125 which are backported here in xonotic2.patch, by reading the FILENAME_DESKTOP_CLIENT and FILENAME_ICON_CLIENT values set in the json `build-options`. This way if upstream wants to make changes such as https://gitlab.com/xonotic/xonotic/-/merge_requests/124#note_2448585470 the changes are contained in that repo so the downstream package build scripts don't need to be patched.